### PR TITLE
New created peers should accept any snapshots.

### PR DESCRIPTION
**Pull request by hicqu**
The PR trys to fix https://github.com/coreos/etcd/issues/9498 with same solution in [TiKV](https://github.com/pingcap/raft-rs/blob/master/src/raft.rs#L1776).

Please take a quickly look at it. If the design is agreed, I will add more test cases for this.

task list:
- [x] Fix the issue.
- [ ] More test cases.

### CODE REVIEW

**Comment by hicqu** [comment]
ping.

**Comment by wenjiaswe** [comment]
cc @jingyih @wenjiaswe

File: raft/raft.go [hunk 1]
```diff
@@ -1270,7 +1270,10 @@ func (r *raft) restore(s pb.Snapshot) bool {
 	}
 
 	// The normal peer can't become learner.
-	if !r.isLearner {
+	//
+	// Both of `prs` and `learnerPrs` are empty means the peer is new created by
+	// conf change, in which case we should accept snapshots make it as learner.
+	if (r.prs.len() > 0 || r.learnerPrs.len() > 0) && !r.isLearner {
```
Comment (xiang90) [inline raft/raft.go]: basically you want to check if r is initialized? if r.initialized() && !r.isLeaner {...} is clearer.

**Comment by hicqu** [comment]
ping @xiang90 PTAL.

**Comment by xiang90** [comment]
can you add more tests? then lgtm.

**Comment by gyuho** [comment]
Superseded by https://github.com/etcd-io/etcd/pull/10154.

**Pull request closed without merging**